lib/sshkit/ssh/channel.ex

```diff
@@ -39,6 +39,31 @@ defmodule SSHKit.SSH.Channel do
     end
   end
 
+  @doc """
+  Activates a subsystem on a channel.
+
+  On success, returns `:ok`. On failure, returns `{:error, reason}`.
+
+  For more details, see [`:ssh_connection.subsystem/4`](http://erlang.org/doc/man/ssh_connection.html#subsystem-4).
+  """
+  @spec subsystem(channel :: struct(), subsystem :: String.t(), options :: list()) ::
+          :ok | {:error, reason :: String.t()}
+  def subsystem(channel, subsystem, options \\ []) do
+    timeout = Keyword.get(options, :timeout, :infinity)
+    impl = Keyword.get(options, :impl, :ssh_connection)
+
+    case impl.subsystem(channel.connection.ref, channel.id, to_charlist(subsystem), timeout) do
+      :success ->
```

Comment:
Super minor, but for the sake of consistency on the code style within the rest of the module, I'd suggest to write these as one line statments:
```elixir
:success -> :ok
err -> err
# etc.
```